tests/includes/legacy/tests-functions.php

```diff
@@ -86,4 +86,30 @@ public function test_give_form_get_default_level() {
 		// When passing invalid form id, it should return null.
 		$this->assertEquals( give_form_get_default_level( 123 ), null );
 	}
+
+    /**
+     * @unreleased
+     * @dataProvider give_donation_form_has_serialized_fields_data
+     */
+    public function test_give_donation_form_has_serialized_fields_returns_true($fields, $expected): void
+    {
+        if ($expected) {
+            $this->assertTrue(give_donation_form_has_serialized_fields($fields));
+        } else {
+            $this->assertFalse(give_donation_form_has_serialized_fields($fields));
+        }
+    }
+
+    /**
+     * @unreleased
+     */
+    public function give_donation_form_has_serialized_fields_data(): array
+    {
+        return [
+            [['foo' => serialize('bar')], true],
+            [['foo' => 'bar'], false],
+            [['foo' => 'bar', 'baz' => serialize('qux')], true],
+            [['foo' => 'bar', 'baz' => 'qux'], false],
```

Comment:
Let's add these new to entries here:

```
[['foo' => 'bar', 'baz' => '\\' . serialize('backslash-bypass')], true],
[['foo' => 'bar', 'baz' => '\\\\' . serialize('double-backslash-bypass')], true],
```

You will see that the last entry will generate an error when running the tests, to solve that check my previous comment.

Reply:
Love it! fd178484